File: problems/125-valid-palindrome/attempt-1-accepted.java
class Solution {
    public boolean isPalindrome(String s) {
        if(s.isEmpty()){
            return true;
        }
        int left =0;
        int right=s.length() - 1;
        while(left<=right){
            char leftMost= s.charAt(left);
            char rightMost =s.charAt(right);
            if(!Character.isLetterOrDigit(leftMost)){
                left++;
            }
            else if (!Character.isLetterOrDigit(rightMost)){
                right --;
            }
            else{
                if(Character.toLowerCase(leftMost)!=Character.toLowerCase(rightMost)){
                   return false; 
                }
                left++;
                right--;
            }
        }
        return true;
        
    }
}

File: problems/9-palindrome-number/attempt-1-accepted.c
bool isPalindrome(int x) {
        
        if (x < 0 || (x && x % 10 == 0))
        {
            return false;
        }
        int a = 0;
        for (; a < x; x /= 10) 
        {
            a = a * 10 + x % 10;
        }
        return x == a || x == a / 10;
    
    
}